libopae/pluginmgr.c

```diff
@@ -71,6 +73,42 @@ STATIC opae_api_adapter_table *adapter_list = (void *)0;
 static pthread_mutex_t adapter_list_lock =
 	PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
 
+
+#define MAX_PLUGINS PLUGIN_SUPPORTED_DEVICES_MAX
+static plugin_cfg plugin_list[MAX_PLUGINS];
+static int plugin_count;
+
+#define CFG_PATHS 4
+static const char *_opae_cfg_files[CFG_PATHS] = {
+	"/etc/opae.cfg",
+	"~/.config/opae/config",
+	"~/.local/opae/config",
+	"~/.local/opae.cfg"
+};
+
+int opae_plugin_mgr_plugin_count(void)
+{
+	return plugin_count;
+}
+
+const plugin_cfg *opae_plugin_mgr_get(int i)
+{
+	return plugin_count ? &plugin_list[i] : NULL;
+}
+
+STATIC const char *find_cfg()
+{
+	int i = 0;
+	struct stat st;
+
+	for (; i < CFG_PATHS; ++i) {
+		if (stat(_opae_cfg_files[i], &st)) {
+			return _opae_cfg_files[i];
```

Comment:
In the case that both /etc/opae.cfg and ~/.local/opae.cfg exist, this will return /etc/opae.cfg.  Can we follow a different model?  Can we process the config file in /etc if any first, then let the local one override settings if it exists?  (similar to what git does, I think.)

Reply:
Longer term, that would be nice. I think let's take small steps, and then iterate.